lib/model/model.go

```diff
@@ -225,6 +226,25 @@ func NewModel(cfg config.Wrapper, id protocol.DeviceID, clientName, clientVersio
 	return m
 }
 
+func (m *model) Stop() {
+	m.Supervisor.Stop()
+	devs := m.cfg.Devices()
+	ids := make([]protocol.DeviceID, 0, len(devs))
+	for id := range devs {
+		ids = append(ids, id)
+	}
+	m.pmut.RLock()
+	closed := make([]chan struct{}, 0, len(m.closed))
+	for _, c := range m.closed {
+		closed = append(closed, c)
+	}
+	m.pmut.RUnlock()
+	m.closeConns(ids, errStopped)
```

Comment:
I am not sure what the strategy here is, but we use error codes to signal stuff, so perhaps it would be worth expanding on that.

Reply:
What strategy? I just explicitly close all connections with an error that essentially says "Bye, not talking to you anymore because I am shutting down".

Reply:
Yeah, but closing connections takes an error, then error types are mapped to error codes in the protocol package which are sent to the other side. (Or that was how it was when I first implemented it)

We are now closing with various types of errors the protocol does know what error code to send for, and they are all probably end up as generic errors with a message. Are there intentions to add separate protocol close codes for this or not?

Reply:
I think you are mixing this up with request responses. Close BEP messages just have a human readable message as field: https://docs.syncthing.net/specs/bep-v1.html#close

Reply:
You are right.